lib/services/discord_rpc.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+import 'dart:core';
+
+import 'package:finamp/services/media_state_stream.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:get_it/get_it.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import 'package:flutter_discord_rpc/flutter_discord_rpc.dart';
+part 'discord_rpc.g.dart';
+
+var running = false;
+StreamSubscription<MediaState>? listener;
+
+@riverpod
```

Comment:
Is there really any reason for making this a provider? We never actually read any data *from* it, do we?  

And you're already using the new `ProviderContainer` too :)

Reply:
I was just copying what ever was done to AutoOffline

Reply:
I believe that is either a) unneeded or b) was done to apply new settings immediately. I think the solution from the playon_service works just as well, without code generation...